aiida/engine/daemon/execmanager.py

```diff
@@ -86,7 +87,7 @@ def upload_calculation(
     computer = node.computer
 
     codes_info = calc_info.codes_info
-    input_codes = [load_node(_.code_uuid, sub_classes=(Code,)) for _ in codes_info]
+    input_codes = [load_node(_.code_uuid, sub_classes=(Code, ContainerCode)) for _ in codes_info]
```

Comment:
Whether it is a sub-class in Python doesn't matter. What matters is the structuring of the entry points, because that is what the query builder uses. So as long as `Code = core.code` and `ContainerCode = core.code.container`, then querying for all `Code`'s will also match all `ContainerCode`'s, unless `subclassing` is explicitly turned off. Note that the `load_node` call here should just be replaced with `load_code` which is the standard for querying for codes.